Getting "readlink: illegal option -- f" when building on macOS

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug



**What happened**:
Make fail with following error
```
$ make all
./hack/build/in-docker.sh "DOCKER_PREFIX= DOCKER_TAG= VERBOSITY= PULL_POLICY= NAMESPACE= ./hack/build/build-manifests.sh"
readlink: illegal option -- f
usage: readlink [-n] [file ...]
make: *** [manifests] Error 1
```
The `readlink` on macOS does not support `--f` option. 
**What you expected to happen**:
CDI should build cleanly.
**How to reproduce it (as minimally and precisely as possible)**:
Execute on macOS machine `make all` 
**Anything else we need to know?**:
Is macOS a support platform for building CDI. I can create a PR to replace using readlink with more platform independent command.

--- a/hack/build/bazel-build-builder.sh
+++ b/hack/build/bazel-build-builder.sh
@@ -15,7 +15,7 @@
 #limitations under the License.
 
 set -e
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/bazel-docker.sh
+++ b/hack/build/bazel-docker.sh
@@ -15,7 +15,7 @@
 #limitations under the License.
 
 set -e
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/build-cdi-func-test-file-host.sh
+++ b/hack/build/build-cdi-func-test-file-host.sh
@@ -16,7 +16,7 @@
 
 set -euo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/build-cdi-func-test-registry-host.sh
+++ b/hack/build/build-cdi-func-test-registry-host.sh
@@ -16,7 +16,7 @@
 
 set -euo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/build-copy-artifacts.sh
+++ b/hack/build/build-copy-artifacts.sh
@@ -16,7 +16,7 @@
 
 set -eo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/build-functest.sh
+++ b/hack/build/build-functest.sh
@@ -14,7 +14,7 @@
 
 set -euo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 
 mkdir -p ${TESTS_OUT_DIR}/

--- a/hack/build/build-go.sh
+++ b/hack/build/build-go.sh
@@ -16,7 +16,7 @@
 
 set -eo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 source "${script_dir}"/version.sh

--- a/hack/build/build-manifests.sh
+++ b/hack/build/build-manifests.sh
@@ -14,7 +14,7 @@
 
 set -euo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/common.sh
+++ b/hack/build/common.sh
@@ -12,7 +12,7 @@
 #See the License for the specific language governing permissions and
 #limitations under the License.
 
-CDI_DIR="$(readlink -f $(dirname $0)/../../)"
+CDI_DIR="$(cd $(dirname $0)/../../ && pwd -P)"
 CDI_GO_PACKAGE=kubevirt.io/containerized-data-importer
 BIN_DIR=${CDI_DIR}/bin
 OUT_DIR=${CDI_DIR}/_out

--- a/hack/build/format.sh
+++ b/hack/build/format.sh
@@ -16,7 +16,7 @@
 
 set -eo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/in-docker.sh
+++ b/hack/build/in-docker.sh
@@ -16,7 +16,7 @@
 
 set -e
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 

--- a/hack/build/resource-generator.sh
+++ b/hack/build/resource-generator.sh
@@ -85,9 +85,9 @@ function processDirTemplates {
 
     templates="$(find "${inTmplPath}" -maxdepth 1 -name "*.in"  -type f)"
     for tmpl in ${templates}; do
-	tmpl_dir="$(cd "$(dirname "${tmpl}")" && pwd -P)"
-	tmpl_filename="$(basename ${tmpl})"
-	tmpl="${tmpl_dir}/${tmpl_filename}"
+        tmpl_dir="$(cd "$(dirname "${tmpl}")" && pwd -P)"
+        tmpl_filename="$(basename ${tmpl})"
+        tmpl="${tmpl_dir}/${tmpl_filename}"
         populateResourceManifest  $generator $outFinalManifestPath $outTmplPath $tmpl $genManifestsDir $outFinalManifestPath
     done
 }

--- a/hack/build/resource-generator.sh
+++ b/hack/build/resource-generator.sh
@@ -14,7 +14,7 @@
 
 set -euo pipefail
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source "${script_dir}"/common.sh
 source "${script_dir}"/config.sh
 
@@ -85,7 +85,9 @@ function processDirTemplates {
 
     templates="$(find "${inTmplPath}" -maxdepth 1 -name "*.in"  -type f)"
     for tmpl in ${templates}; do
-        tmpl=$(readlink -f "${tmpl}")
+	tmpl_dir="$(cd "$(dirname "${tmpl}")" && pwd -P)"
+	tmpl_filename="$(basename ${tmpl})"
+	tmpl="${tmpl_dir}/${tmpl_filename}"
         populateResourceManifest  $generator $outFinalManifestPath $outTmplPath $tmpl $genManifestsDir $outFinalManifestPath
     done
 }

--- a/hack/build/run-functional-tests.sh
+++ b/hack/build/run-functional-tests.sh
@@ -19,7 +19,7 @@ set -euo pipefail
 readonly MAX_CDI_WAIT_RETRY=30
 readonly CDI_WAIT_TIME=10
 
-script_dir="$(readlink -f $(dirname $0))"
+script_dir="$(cd "$(dirname "$0")" && pwd -P)"
 source hack/build/config.sh
 source hack/build/common.sh
 

--- a/hack/update-codegen.sh
+++ b/hack/update-codegen.sh
@@ -20,7 +20,7 @@ set -o pipefail
 
 export GO111MODULE=off
 
-SCRIPT_ROOT=$(readlink -f $(dirname ${BASH_SOURCE})/..)
+SCRIPT_ROOT="$(cd "$(dirname $0)/../" && pwd -P)"
 CODEGEN_PKG=${CODEGEN_PKG:-$(
     cd ${SCRIPT_ROOT}
     ls -d -1 $GOPATH/src/k8s.io/code-generator 2>/dev/null || echo ../code-generator

--- a/hack/version/set-version.sh
+++ b/hack/version/set-version.sh
@@ -3,7 +3,7 @@
 set -eou pipefail
 
 # containerized-data-importer/
-REPO_ROOT="$(readlink -f $(dirname $0)/../../)"
+REPO_ROOT="$(cd "$(dirname $0)/../../")" && pwd -P)"
 
 source $REPO_ROOT/hack/version/lib.sh
 

